bgpd/bgp_mplsvpn.c

```diff
@@ -1336,6 +1336,228 @@ leak_update(struct bgp *to_bgp, struct bgp_dest *bn,
 	return new;
 }
 
+void bgp_mplsvpn_path_nh_label_unlink(struct bgp_path_info *path)
+{
+	struct bgp_label_per_nexthop_cache *blnc;
+
+	if (!path)
+		return;
+
+	blnc = path->label_nexthop_cache;
+
+	if (!blnc)
+		return;
+
+	LIST_REMOVE(path, label_nh_thread);
+	path->label_nexthop_cache->path_count--;
+	path->label_nexthop_cache = NULL;
+
+	if (LIST_EMPTY(&(blnc->paths)))
+		bgp_label_per_nexthop_free(blnc);
+}
+
+static void
+bgp_mplsvpn_path_nh_label_update(struct bgp_label_per_nexthop_cache *blnc)
```

Comment:
`bgp_mplsvpn_path_nh_label_update() i`s used only once

Reply:
removed the call, and added the code in the parent function